app/controllers/ApplicationController.scala

```diff
@@ -507,48 +505,38 @@ case class ApplicationController @Inject() (
       webJarsUtil: org.webjars.play.WebJarsUtil,
       request: RequestWithUserData[A]
   ): Future[Html] = {
-
-    val (usersFuture, applicationsFutureNoDateFilter, groupsFuture) =
-      if (areaIds.isEmpty && organisationIds.isEmpty && groupIds.isEmpty) {
-        (userService.allNoNameNoEmail, applicationService.all(), userGroupService.all)
-      } else if (areaIds.nonEmpty && groupIds.isEmpty) {
-        val groupsFuture = userGroupService.byAreas(areaIds)
-        if (organisationIds.isEmpty) {
-          val usersFuture = groupsFuture.flatMap { groups =>
-            val groupIds = groups.map(_.id)
-            userService.byGroupIdsAnonymous(groupIds)
-          }
-          (usersFuture, applicationService.allForAreas(areaIds), groupsFuture)
-        } else {
-          val groupsFuture = userGroupService.byOrganisationIds(organisationIds).map { groups =>
-            groups.filter(group => group.areaIds.intersect(areaIds).nonEmpty)
-          }
-          val usersFuture =
-            groupsFuture.flatMap(groups => userService.byGroupIdsAnonymous(groups.map(_.id)))
-          val applicationsFuture = usersFuture
-            .flatMap(users => applicationService.allForUserIds(users.map(_.id)))
-            .map(_.filter(application => areaIds.contains(application.area)))
-          (usersFuture, applicationsFuture, groupsFuture)
-        }
-      } else if (organisationIds.nonEmpty) {
-        val groupsFuture = userGroupService.byOrganisationIds(organisationIds)
-        val usersFuture =
-          groupsFuture.flatMap(groups => userService.byGroupIdsAnonymous(groups.map(_.id)))
-        val applicationsFuture =
-          usersFuture.flatMap(users => applicationService.allForUserIds(users.map(_.id)))
-        (usersFuture, applicationsFuture, groupsFuture)
-      } else {
-        val groupsFuture = userGroupService.byIdsFuture(groupIds)
-        val usersFuture =
-          groupsFuture.flatMap(groups => userService.byGroupIdsAnonymous(groups.map(_.id)))
-        val applicationsFuture =
-          usersFuture.flatMap(users => applicationService.allForUserIds(users.map(_.id)))
-        (usersFuture, applicationsFuture, groupsFuture)
+    val usersAndApplications =
```

Comment:
Avoir un type ici aiderait beaucoup à la relecture, je suppose `Future[(List[User], List[Application])]` ?
```suggestion
    val usersAndApplications: Future[(List[User], List[Application])] =
```